### ISSUE TITLE
Using message key in new-style logging overrides new-style formatting.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7855]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)| @ralphm reported|
|-|-|
|Trac ID|trac#7855|
|Type|defect|
|Created|2015-04-15 19:27:24Z|
|Branch|https://github.com/twisted/twisted/tree/log-message-override-7855|

When new style-logging is used in combination with old-style observers, the use of a key `message` should not override the new-style formatting through `log_format`.

```python
import sys
from twisted.python import log
from twisted.logger import Logger

log.startLogging(sys.stderr)

logger = Logger()
logger.info("something {message}", message="foo")
```

Should render as `something foo`, not `foo`.

<details><summary>Searchable metadata</summary>

```
trac-id__7855 7855
type__defect defect
reporter__ralphm ralphm
priority__normal normal
milestone__New_Logging_System New Logging System
branch__branches_log_message_override_7855 branches/log-message-override-7855
branch_author__ralphm ralphm
status__closed closed
resolution__fixed fixed
component__logger logger
keywords__BOOYAH_ BOOYAH!
time__1429126044013929 1429126044013929
changetime__1481165881696493 1481165881696493
version__None None
owner__wsanchez wsanchez

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_1"></a>|@ralphm commented|
|-|-|

(In [44468]) Branching to 'log-message-override-7855'

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)|@ralphm set owner to @wsanchez|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_3"></a>|@wsanchez commented|
|-|-|

Yup, awesome.

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_4"></a>|@ralphm set status to `closed`|
|-|-|

(In [44474]) Merge log-message-override-7855: Correctly format log event with key named `message`.

Author: ralphm
Reviewer: wsanchez
Fixes: [#7855](https://github.com/twisted/twisted/issues/7855)
